tests/cypress/support/helpers/test-connector.ts

```diff
@@ -0,0 +1,46 @@
+import { createWalletClient, http, type Chain, type Hex } from 'viem'
+import { privateKeyToAccount } from 'viem/accounts'
+import { createConnector, type CreateConnectorFn } from '@wagmi/core'
+
+type Options = {
+  /** A hexadecimal private key used to generate a test account */
+  privateKey: Hex
+  /** The testnet chain configuration */
+  chain: Chain
+}
+
+/**
+ * Cypress test connector for Wagmi.
+ *
+ * This connector is designed for use in test environments (e.g., Cypress) with a testnet chain.
+ * It creates a wallet using a custom seed (private key) to allow testing contract write calls
+ * without relying on third-party browser extensions like MetaMask.
```

Comment:
Did you try just injecting `window.ethereum` instead? It sounds simpler to use the injected provider

Reply:
Sounds like a really roundabout way of achieving the same result. I think using a connector is the way to go. Wagmi also provides a [mock](https://wagmi.sh/core/api/connectors/mock) connector but that's of course not what we want here. They used to have docs that suggest creating connectors but it seems to have vanished...

Reply:
No problem. FYI window.ethereum interface is [super simple](https://docs.metamask.io/wallet/reference/provider-api/)